packages/boilerplate-generator/generator.js

```diff
@@ -23,13 +26,30 @@ export class Boilerplate {
   // purpose is to allow you to specify data that you might not know at
   // the time that you construct the Boilerplate object. (e.g. it is used
   // by 'webapp' to specify data that is only known at request-time).
+  // this returns a stream
   toHTML(extraData) {
-    if (!this.baseData || !this.template) {
+    if (!this.baseData || !this.headTemplate || !this.closeTemplate) {
       throw new Error('Boilerplate did not instantiate correctly.');
     }
 
-    return  "<!DOCTYPE html>\n" +
-      this.template({ ...this.baseData, ...extraData });
+    const data = {...this.baseData, ...extraData};
+    const start = "<!DOCTYPE html>\n" + this.headTemplate(data);
+
+    const { body, dynamicBody } = data;
+
+    const end = this.closeTemplate(data);
+    const response = stream.create();
+    
+    response.append(Buffer.from(start))
+    if (body) {
+      response.append(typeof body === "string" ? Buffer.from(body) : body)
```

Comment:
What else can `body` be? Let's enforce our assumptions here using some combination of `typeof body === "string"`, `Buffer.isBuffer`, and `typeof body.read === "function"` (if that covers all the cases).

Reply:
The internet tells me `body instanceof ReadableStream` isn't always safe, since folks can implement their own stream-like objects, which is why I suggested the dynamic duck test that `body.read` is a function. Maybe extract this logic into a helper function so you can do `appendToStream(response, body)`?